### DISCUSSION TITLE
Self Hosted Docker runner targeting self hosted Gitlab

### DISCUSSION THREAD

**Discussion by gaeljw** [renovatebot/renovate#13691] [Request Help]
Hello,

It took me a bit of time to understand how to setup Renovate when running via the Docker image (self hosted) to target a self-hosted Gitlab.

I'm wondering If i missed something in the documentation or if we should add it somewhere.

I was looking for a piece of documentation telling me that I had to do something like this:
```
docker run --rm -v "$(pwd)/renovate.js:/usr/src/app/config.js" \
    -e "GITHUB_COM_TOKEN=any-personal-user-token-for-github-com-for-fetching-changelogs" \
    -e "RENOVATE_TOKEN=self-hosted-gitlab-renovate-user-token" \
    -e "RENOVATE_PLATFORM=gitlab" \
    -e "RENOVATE_ENDPOINT=https://self-hosted.gitlab/api/v4/" \
    renovate/renovate
```

**Comment by viceice** [comment]
Checkout our official gitlab pipeline templates at https://gitlab.com/renovate-bot/renovate-runner

**Reply by gaeljw** [reply]
Indeed there's a good portion of information here but as I'm not using Gitlab pipelines but running the docker container standalone from an external system, it's not the place where I looked into.

Maybe what is missing is a exhaustive list of all environment variables needed to run Renovate and in which cases they are automatically provided (like Gitlab URL and platform is inferred when running from a folder tracked by Git)?

**Reply by viceice** [reply]
https://github.com/renovatebot/docker-renovate the minimal slim version

**Reply by viceice** [reply]
https://docs.renovatebot.com/getting-started/running/#self-hosting-renovate
